extension/src/popup/views/SignTransaction/index.tsx

```diff
@@ -60,11 +102,45 @@ export const SignTransaction = () => {
   // the public key the user had selected before starting this flow
   const defaultPublicKey = useRef(publicKey);
   const allAccountsMap = useRef({} as { [key: string]: Account });
-  const accountToSign = blobOrTx.accountToSign; // both types have this key
+
+  const tx = getTransactionInfo(location.search);
+
+  const {
+    accountToSign: _accountToSign,
+    transactionXdr,
+    domain,
+    isDomainListedAllowed,
+    isHttpsDomain,
+    flaggedKeys,
+  } = tx;
+
+  /* 
+  Reconstruct the tx from xdr as passing a tx through extension contexts 
+  loses custom prototypes associated with some values. This is fine for most cases 
+  where we just need a high level overview of the tx, like just a list of operations.  
+  But in this case, we will need the hostFn prototype associated with Soroban tx operations.
+  */
+
+  const SDK = isExperimentalModeEnabled ? SorobanSdk : StellarSdk;
```

Comment:
Wait no, `StellarSdk` is [`js-stellar-sdk`](https://github.com/stellar/js-stellar-sdk) which connects to Horizon.

Reply:
ah whoops, thanks for catching that. I thin the rest of it still makes sense but let me know if you think otherwise.

Reply:
hold up, I'm back to my original statement that we do not need to do this because we only use an sdk in this workflow to rebuild the tx from xdr with `TransactionBuilder` which is available and the same through either sdk so we should just use `SorobanSdk` here.

Reply:
oh sure yeah, that makes sense. but if you guys need horizon as part of freighter beyond experimental/futurenet mode then you may need both (until https://github.com/stellar/js-stellar-sdk/issues/850 lands)

Reply:
ah nice I'll watch that one. I think we mostly did this in anticipation and don't really need to do it many of the places that we do.